domain/machine/state/state.go

```diff
@@ -1138,6 +1005,42 @@ WHERE uuid = $machineUUID.uuid
 	return result, nil
 }
 
+// GetMachineContainers returns the names of the machines which have as parent
+// the specified machine.
+func (st *State) GetMachineContainers(ctx context.Context, mUUID string) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	ident := machineUUID{UUID: mUUID}
+	query := `
+SELECT &machineName.*
+FROM machine
+JOIN machine_parent ON machine.uuid = machine_parent.machine_uuid
+WHERE parent_uuid = $machineUUID.uuid`
+	queryStmt, err := st.Prepare(query, machineName{}, ident)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	var results []machineName
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, queryStmt, ident).GetAll(&results)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
```

Comment:
Should we have a check for non-existent machines first? In that case your code currently returns an empty slice whilst it should return an error.